plugins/node/opentelemetry-instrumentation-cassandra/src/instrumentation.ts

```diff
@@ -0,0 +1,366 @@
+
+          const patchedCallback = function (
+            this: unknown,
+            ...cbArgs: Parameters<typeof originalCallback>
+          ) {
+            const error = cbArgs[0];
+
+            if (error) {
+              span.setStatus({
+                code: SpanStatusCode.ERROR,
+                message: error.message,
+              });
+              span.recordException(error);
+            }
+
+            span.end();
+
+            return originalCallback.apply(this, cbArgs);
+          };
+
+          args[args.length - 1] = patchedCallback;
+
+          return context.with(batchContext, () => {
+            return original.apply(this, args);
+          });
+        }
+
+        const batchPromise = safeExecuteInTheMiddle(
+          () => {
+            return original.apply(
+              this,
+              args
+            ) as Promise<CassandraDriver.types.ResultSet>;
+          },
+          error => {
+            if (error) {
+              failSpan(span, error);
+            }
+          }
+        );
+
+        const wrappedPromise = wrapPromise(span, batchPromise);
+
+        return context.bind(batchContext, wrappedPromise);
+      };
+    };
+  }
+
+  private _getPatchedStream() {
+    return (original: (...args: unknown[]) => EventEmitter) => {
+      const plugin = this;
+      return function patchedStream(
+        this: CassandraDriver.Client,
+        ...args: unknown[]
+      ) {
+        // Since stream internally uses execute, there is no need to add DB_STATEMENT twice
+        const span = plugin.tracer.startSpan('cassandra-driver.stream', {
+          kind: SpanKind.CLIENT,
+          attributes: {
+            [SemanticAttributes.DB_SYSTEM]: DbSystemValues.CASSANDRA,
+          },
+        });
+
+        const callback = args[3];
+
+        const endSpan = (error: Error) => {
+          if (error) {
+            span.setStatus({
+              code: SpanStatusCode.ERROR,
+              message: error.message,
+            });
+            span.recordException(error);
+          }
+          span.end();
+        };
+
+        if (callback === undefined) {
+          args[3] = endSpan;
+        } else if (typeof callback === 'function') {
+          const wrappedCallback = function (this: unknown, err: Error) {
+            endSpan(err);
+            return callback.call(this, err);
+          };
+          args[3] = wrappedCallback;
+        }
+
+        return safeExecuteInTheMiddle(
+          () => {
+            return original.apply(this, args);
+          },
+          error => {
+            if (error) {
+              failSpan(span, error);
+            }
+          }
+        );
+      };
+    };
+  }
+}
+
+function startSpan(
+  tracer: Tracer,
+  op: string,
+  statement: string,
+  client: CassandraDriver.Client
+) {
+  const attributes = {
+    [SemanticAttributes.DB_SYSTEM]: DbSystemValues.CASSANDRA,
+    [SemanticAttributes.DB_STATEMENT]: statement,
```

Comment:
we generally do not add the statement by default since it can contains sensitive values, could add `enhancedDatabaseReporting` like mongodb (https://github.com/open-telemetry/opentelemetry-js-contrib/blob/621f0141e406b75ef11a77fb85ce8caab9c8a272/plugins/node/opentelemetry-instrumentation-mongodb/src/types.ts#L25) ?

Reply:
Disabled it by default and added the same option